src/app/_components/userProfile/UserProfile.tsx

```diff
@@ -1,19 +1,53 @@
-'use client'
+import QRCode from 'react-qr-code'
+import { TRPCError } from '@trpc/server'
+import SignOut from '~/app/_components/userProfile/signOut'
+import { api } from '~/trpc/server'
+import { getServerAuthSession } from '~/server/auth'
 
-import type { Session } from 'next-auth'
+export default async function UserProfile() {
+  const session = await getServerAuthSession()
 
-interface UserProfileProps {
-  session: Session
-}
+  if (!session) {
+    throw new TRPCError({
+      code: 'NOT_FOUND',
+      message: `Session not found`,
+    })
+  }
+
+  const team = await api.team.getByUserId({ userId: session.user.id })
+
+  if (!team) {
+    throw new TRPCError({
+      code: 'NOT_FOUND',
+      message: `Team not found`,
+    })
+  }
 
-export function UserProfile(props: UserProfileProps) {
   return (
     <div>
       <h1>Your application is just your profile.</h1>
       <h2>All you have to do is fill it in.</h2>
-      <pre>
-        {JSON.stringify(props.session, null, 2)}
-      </pre>
+      {JSON.stringify(session, null, 2)}
+
+      <h1>Your team</h1>
+      <p>
+        Team name:
+        {team.name}
+      </p>
+      <p>
+        Invite code:
+        {team.id}
+      </p>
+
+      <p>Share this QR code with your friends to invite them!</p>
+      <QRCode
+        size={256}
+        style={{ height: 'auto', maxWidth: '15%', width: '15%', margin: '2rem' }}
+        value={`http://localhost:8000/t/j/${team.id}`}
```

Comment:
Also are these ids hashed? (i.e. do they look like '123' or are they 'xFjdgaor')? Because I think it's a problem if they aren't hashed. 

If they aren't hashed then someone could just iterate through all the teams and users and pull all their names just by scraping our site.

Reply:
They're hashed 😉

https://github.com/cuhacking/hackathon/blob/df87210da03423c5d24ae8c2ede57f3c847e01c4/src/server/auth.ts#L60-L62

https://github.com/cuhacking/hackathon/blob/df87210da03423c5d24ae8c2ede57f3c847e01c4/src/server/auth.ts#L76-L82